linera-execution/src/evm/database.rs

```diff
@@ -67,6 +67,7 @@ impl StorageStats {
 
 pub(crate) struct DatabaseRuntime<Runtime> {
     storage_stats: Arc<Mutex<StorageStats>>,
+    pub contract_address: Address,
```

Comment:
so this is an address of a contract that is currently being executed?

Reply:
No to the rename:
* `current_address` makes it look like it will change. It will not.
* `current_address` hides the fact that this is the address of the contract

Instead, I added comments on those points.

Reply:
It is because you cannot access to the runtime when building the new. Otherwise I would have implemented it that way.

When putting the `Address::ZERO`, we are not putting a valid address to the contract, we put something because we have to. The correct value is set to by the call of `set_contract_address`.

Therefore, it is incorrect to say that it that the contract address changes, it is being set.

Reply:
Current comment are useless. A variable `storage_stats` got a comment "This is the storage statistics.". A variable `contract_address` got a comment "This is the EVM address of the contract". What address? Current? prevous? caller? Callee? Yours, mine? Write commens such that they help the reader - that is the rule #2 of writing comments (rule #1 being don't comment what can be expressed by types or variable names).

There is no comment on `DatabaseRuntime` so reader has no idea what's the context.

The `contract_address` here is the address of contract being currently executed/on the stack. It can change - because it starts as `Address.ZERO`.